lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py

```diff
@@ -1671,6 +1672,41 @@ def get_url(self, subsection_id=None, user_id=None):  # pylint: disable=argument
         )
         return "{0}?user_id={1}".format(base_url, user_id or self.user_id)
 
+    @ddt.data(
+        'login_staff',
+        'login_course_admin',
+        'login_course_staff',
+    )
+    def test_no_grade(self, login_method):
+        getattr(self, login_method)()
+        user_no_grade = UserFactory.create()
+
+        with self.assertRaises(PersistentSubsectionGrade.DoesNotExist):
+            PersistentSubsectionGrade.objects.get(
+                user_id=user_no_grade.id,
+                course_id=self.usage_key.course_key,
+                usage_key=self.usage_key
+            )
+
+        resp = self.client.get(
+            self.get_url(subsection_id=self.usage_key, user_id=user_no_grade.id)
+        )
+
+        expected_data = {
+            'original_grade': OrderedDict([
+                ('earned_all', 0.0),
+                ('possible_all', 0.0),
```

Comment:
Is there a combination of mocks we can put together that would capture the case where there is a possible score to be calculated? I'm not sure it makes as much sense to test this situation if we're not capturing the situation we're interested in modeling

Reply:
I agree.  I also find it weird that the expected values are 0 here, since the test below this, `test_no_override`, shows non-zero scores for the same test subsection.

Reply:
The test_no_grade has a user created specifically for it without a grade.

Reply:
Right, but should they have >0 `possible_graded` points?  That's kind of the point of this whole PR - even if a student hasn't attempted the subsection yet, we still want to see how many points are possible for them to earn in the subsection.

Reply:
Understood. I am having a bit of difficulty coming up with a unit test. However, it works fine in integration testing on my local environment. However, I think we may need to do a tweak in gradebook to actually display it.

Reply:
unit test updated.